### PR TITLE
updated test strategy document

### PR DESCRIPTION
**Pull request by Faizi52996**
updated test-strategy document with the sequence for running the automated tests.

### CODE REVIEW

File: documentation/testing/test-strategy.md [hunk 1]
```diff
@@ -89,54 +92,67 @@ be requested as appropriate.
...
+
+[Xunit](https://github.com/xunit)
 
 #### Feature/E2E Testing
 
 Feature and end-to-end tests are both designed to simulate the behaviour of an end user. Tests are written from an end-user perspective and run
 against deployed environments through a web-browser (such as Chrome, Safari, Firefox etc.) driven by web-drivers. This is the slowest type of (functional) automated
 testing are run later than unit and integration tests (see testing pyramid).
 
+These tests will run in testing environment after build stage in Testing environment. The testing environment will have pre loaded data to be used by tests.
```
Comment (Chris-Lorch) [inline documentation/testing/test-strategy.md:118]: Can you remind me where this testing environment is if we are using trunk based development? Do we have a separate testing environment set up, or is the current stage of the `main` branch essentially the testing enviroment?

File: documentation/testing/test-strategy.md [hunk 2]
```diff
@@ -89,54 +92,67 @@ be requested as appropriate.
...
 
-#### API Testing
 API tests are designed to test that the correct responses are received from the API when requests are made to it. These tests are run as part of the fuctional testing
 which runs as part of the pipeline with each deployment. These tests are written by collaboration of dev and test engineer.
+We have 2 types of API Tests. 
 
-##### Technology Selected
+#### API Integration Test
+Same as Web Application Integrations tests these tests are the responsibility of the developers to write and maintain in collaboration with the test engineers.
+In the pipeline the tests are kicked off in build stage after code changes in platform project.
```
Comment (Chris-Lorch) [inline documentation/testing/test-strategy.md:135]: suggested edit: "...changes in the platform project."

File: documentation/testing/test-strategy.md [hunk 3]
```diff
@@ -89,54 +92,67 @@ be requested as appropriate.
...
 
-#### Performance Testing
-Performance testing is used to evaluate how the service is able to cope in terms of its stability, responsiveness and usability when put under different simulated workloads.
-This ensures that the service copes with large amount of users accessing the service at one time.
+- [Moq](https://learn.microsoft.com/en-us/shows/visual-studio-toolbox/unit-testing-moq-framework)
 
+- [Xunit](https://github.com/xunit)
 
+#### API Functional Tests
+ These tests checks the functionality and behaviour of deployed APIs. These tests are kicked off after build stage using the test environment after code changes are made.
```
Comment (Chris-Lorch) [inline documentation/testing/test-strategy.md:144]: suggested edit:
"... after the build stage using...."

File: documentation/testing/test-strategy.md [hunk 4]
```diff
@@ -89,54 +92,67 @@ be requested as appropriate.
...
-** *add technology here* **
+- [Xunit](https://github.com/xunit)
+- [Specflow](https://specflow.org/)
+
 
 #### Accessibility Testing
 Automated accessibility testing will be carried out using [Deque.axe.playwright](https://github.com/dequelabs/axe-core-nuget/blob/develop/packages/playwright/README.md)
 library to generate feedback on whether the service meets high level accessibility requirements in line with WCAG 2.2 AA.
-This will run as part of the pipeline in the later deployment steps from development to test and production.
+These tests will run every weekend in a separate pipeline using the testing environment.
```
Comment (Chris-Lorch) [inline documentation/testing/test-strategy.md:155]: Do we have a specific time they are expected to run, and a justification for that time? What impact, if any, could this have on the developers / users?

**Review by Chris-Lorch — COMMENTED** [review]
Just some minor comments

File: documentation/testing/test-strategy.md (hunk 1 above)
Comment (Faizi52996) [reply documentation/testing/test-strategy.md:118]: That's the decision we need to make to setup a separate environment for testing. I have marked this PR as draft until we have concluded what we are going to do.

**Pull request closed without merging**